R/importance_sampling.R

```diff
@@ -0,0 +1,209 @@
+
+#' @rdname psis
+importance_sampling <- function(log_ratios, ...) UseMethod("importance_sampling")
+
+#' @noRd
+#' @description
+#' Currently implemented importance sampling methods
+implemented_is_methods <- function() c("PSIS", "TIS", "SIS")
+
+#' @rdname psis
+importance_sampling.array <-
+  function(log_ratios, ...,
+           r_eff = NULL,
+           cores = getOption("mc.cores", 1),
+           is_method) {
+    cores <- loo_cores(cores)
+    stopifnot(length(dim(log_ratios)) == 3)
+    stopifnot(is_method %in% implemented_is_methods())
+    log_ratios <- validate_ll(log_ratios)
+    log_ratios <- llarray_to_matrix(log_ratios)
+    r_eff <- prepare_psis_r_eff(r_eff, len = ncol(log_ratios))
+    do_importance_sampling(log_ratios, r_eff = r_eff, cores = cores, is_method = is_method)
+  }
+
+#' @rdname psis
+importance_sampling.matrix <-
+  function(log_ratios,
+           ...,
+           r_eff = NULL,
+           cores = getOption("mc.cores", 1),
+           is_method) {
+    cores <- loo_cores(cores)
+    stopifnot(is_method %in% implemented_is_methods())
+    log_ratios <- validate_ll(log_ratios)
+    r_eff <- prepare_psis_r_eff(r_eff, len = ncol(log_ratios))
+    do_importance_sampling(log_ratios, r_eff = r_eff, cores = cores, is_method = is_method)
+  }
+
+#' @rdname psis
+importance_sampling.default <-
```

Comment:
When I run `loo(fit, is_method = "SIS")` with a brms model for instance and the new loo version, the output is still a `psis_loo` object that tells me that all khat were good although I don't think that SIS actually computed khats at all.

Reply:
Agree. this is a bug. Will fix.

Reply:
Fixed